### PR TITLE
Implement `CrossFitEstimator.score`

### PR DESCRIPTION
**Pull request by FrancescMartiEscofetQC**
This PR is just for the sake of completeness.
# Checklist

- [x] Added a `CHANGELOG.rst` entry

### CODE REVIEW

File: metalearners/cross_fit_estimator.py [hunk 1]
```diff
@@ -337,8 +338,28 @@ def predict_proba(
...
+        sample_weight: Vector | None = None,
+    ) -> float:
+        """Return the coefficient of determination of the prediction if the estimator is
+        a regressor or the mean accuracy if it is a classifier."""
+        if self._estimator_type == "classifier":
+            return accuracy_score(
+                y, self.predict(X, is_oos, oos_method), sample_weight=sample_weight
+            )
+        elif self._estimator_type == "regressor":
+            return r2_score(
```
Comment (kklein) [inline metalearners/cross_fit_estimator.py:356]: Any particular reason why you chose R2, rather than e.g. the RMSE?

File: metalearners/cross_fit_estimator.py [hunk 2]
```diff
@@ -337,8 +338,28 @@ def predict_proba(
...
+        self,
+        X: Matrix,
+        y: Vector,
+        is_oos: bool,
+        oos_method: OosMethod | None = None,
+        sample_weight: Vector | None = None,
+    ) -> float:
+        """Return the coefficient of determination of the prediction if the estimator is
+        a regressor or the mean accuracy if it is a classifier."""
+        if self._estimator_type == "classifier":
```
Comment (kklein) [inline metalearners/cross_fit_estimator.py:351]: Do you happen to know what the trade-off between doing this and using `sklearn`'s `is_classifier` is here?

E.g. https://github.com/Quantco/metalearners/blob/9e00603d5b16d03981294a9d3c7f74393d41ad13/metalearners/cross_fit_estimator.py#L165

File: metalearners/cross_fit_estimator.py (hunk 1 above)
Comment (FrancescMartiEscofetQC) [reply metalearners/cross_fit_estimator.py:356]: Just because it looks like it is the convention in ``sklearn``, see [this](https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.LGBMRegressor.html#lightgbm.LGBMRegressor.score) and [this](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html#sklearn.linear_model.LinearRegression.score).
I think they do it because the R2 has as known range whether the RMSE depends a lot on the data but I just followed their rule, R2 for regressors and accuracy for classifiers.

File: metalearners/cross_fit_estimator.py (hunk 2 above)
Comment (FrancescMartiEscofetQC) [reply metalearners/cross_fit_estimator.py]: None, I just didn't think about it 😅 . I think it's better to use the `sklearn` methods.
[108f4f3](https://github.com/Quantco/metalearners/pull/63/commits/108f4f381b5273635995b6f68d37d33244276cc9)
